src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -204,23 +204,63 @@ impl<T: 'static> EventLoop<T> {
             // empty buffer of events
             {
                 let mut guard = sink.lock().unwrap();
-                guard.empty_with(|evt| callback(evt, &self.window_target, &mut control_flow));
+                guard.empty_with(|evt| {
+                    // make ControlFlow::Exit sticky by providing a dummy
+                    // control flow reference if it is already Exit.
+                    let mut dummy = ControlFlow::Exit;
```

Comment:
Can we avoid the code duplication by separating this out in a function. I wrote a small patch: https://termbin.com/lq9d